GVFS/GVFS.Tests/Should/EnumerableShouldExtensions.cs

```diff
@@ -48,7 +48,7 @@ public static T ShouldContainSingle<T>(this IEnumerable<T> group, Func<T, bool>
         public static void ShouldNotContain<T>(this IEnumerable<T> group, Func<T, bool> predicate)
         {
             T item = group.SingleOrDefault(predicate);
-            item.ShouldEqual(default(T));
+            item.ShouldEqual(default(T), "Unexpected matching entry found in {" + string.Join(",", group) + "}");

```

Comment:
I think the body of this method *could* be simplified to:

 `group.Any(predicate).ShouldEqual(false, "Unexpected matching entry found in {" + string.Join(",", group) + "}"); `

Reply:
I think I'll leave this as-is for now and keep the changes focused on fixing the functional test reliability issue (I do agree this is a nice potential cleanup, but would rather not use this PR for refactoring\cleaning the `Should` helpers)